src/System.Memory/tests/Span/Indexer.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Numerics;
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class IndexerTests
+    {
+        [Fact]
+        public static void IndexerWithIndexTest()
+        {
+            ReadOnlySpan<char> span = "Hello".AsSpan();
+            Assert.Equal('H', span[new Index(0, fromEnd: false)]);
+            Assert.Equal('o', span[new Index(1, fromEnd: true)]);
+            Assert.Equal(span[new Index(2, fromEnd: false)], span[new Index(span.Length - 2, fromEnd: true)]);
+
+            Assert.Throws<IndexOutOfRangeException>(() =>
+            {
+                char c;
+                ReadOnlySpan<char> s = "Hello".AsSpan();
+                c = s[new Index(0, fromEnd: true)];
+                return;
+            });
```

Comment:
You could make this whole thing a one-liner with:
```C#
Assert.Throws<IndexOutOfRangeException>(() => "Hello".AsSpan()[new Index(0, fromEnd: true)]);
```
Same for other similar cases.